moveit_ros/moveit_servo/include/moveit_servo/servo_calcs.h

```diff
@@ -77,6 +77,28 @@ enum class ServoType
   JOINT_SPACE
 };
 
+/** \brief Possibly calculate a velocity scaling factor, due to proximity of
+ * singularity and direction of motion
+ * @param[in] joint_model_group   The MoveIt group
+ * @param[in] commanded_twist     The commanded Cartesian twist
+ * @param[in] svd                 A singular value decomposition of the Jacobian
+ * @param[in] pseudo_inverse      The pseudo-inverse of the Jacobian
+ * @param[in] hard_stop_singularity_threshold  Halt if condition(Jacobian) > hard_stop_singularity_threshold
+ * @param[in] lower_singularity_threshold      Decelerate if condition(Jacobian) > lower_singularity_threshold
+ * @param[in] leaving_singularity_threshold_multiplier      Allow faster motion away from singularity
+ * @param[in, out] clock          A ROS clock, for logging
+ * @param[in, out] current_state  The state of the robot. Used in internal calculations.
+ * @param[out] status             Singularity status
+ */
+double velocityScalingFactorForSingularity(const moveit::core::JointModelGroup* joint_model_group,
```

Comment:
This was changed to a free function, to make testing easier. All of the inputs and outputs are specified explicitly now

Reply:
You can move it to a separate file then if you'd like, that would help a small amount with the compile time issues of servo_calcs.cpp.o.

Reply:
done